From 4273d800091d45c5e6ecee25457c5d4387282adf Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Fri, 8 Nov 2024 16:01:32 +0100
Subject: [PATCH 1351/1695] dts: stm32: fix memory ranges on fmc node for mp231

Fix memory ranges on fmc node for mp231.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 00acb1b7c1f7df61afaa72cb0179a70ebeee2a1f dts: stm32: fix memory ranges on fmc node for mp231]
Change-Id: Ic046c38c8f70fc6635a9481bb9b8d1bb77680e71
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/417514
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 9d4e362d8..804f2b93d 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -3246,19 +3246,18 @@
 		};
 
 		fmc: memory-controller@48200000 {
-			#address-cells = <2>;
-			#size-cells = <1>;
 			compatible = "st,stm32mp25-fmc2-ebi";
 			reg = <0x48200000 0x400>;
+			ranges = <0 0 0x70000000 0x04000000>, /* EBI CS 1 */
+				 <1 0 0x74000000 0x04000000>, /* EBI CS 2 */
+				 <2 0 0x78000000 0x04000000>, /* EBI CS 3 */
+				 <3 0 0x7c000000 0x04000000>, /* EBI CS 4 */
+				 <4 0 0x48810000 0x00001000>; /* NAND */
+			#address-cells = <2>;
+			#size-cells = <1>;
 			clocks = <&rcc CK_KER_FMC>;
 			resets = <&rcc FMC_R>;
 			status = "disabled";
-
-			ranges = <0 0 0x60000000 0x04000000>, /* EBI CS 1 */
-				 <1 0 0x64000000 0x04000000>, /* EBI CS 2 */
-				 <2 0 0x68000000 0x04000000>, /* EBI CS 3 */
-				 <3 0 0x6c000000 0x04000000>, /* EBI CS 4 */
-				 <4 0 0x80000000 0x10000000>; /* NAND */
 		};
 	};
 };
-- 
2.47.2

